### PR TITLE
Add minecraft-quilt template

### PR DESCRIPTION
**Pull request by rickiewars**
Solves issue #143

### CODE REVIEW

**Comment by rickiewars** [comment]
I've successfully been able to create a new quilt server inside my pufferpanel instance of Docker.
Quilt is an upcomming successor of Fabric and I'm seeing mods dropping support for Fabric in fafor of Quilt.

**Comment by LordRalex** [comment]
Please rebase this PR to v2.6 and follow our formatting rules:
https://github.com/PufferPanel/templates/blob/v2.6/README.md#formattng-rules

Please also confirm there is not a corrected template in the new branch.

**Comment by rickiewars** [comment]
> Please rebase this PR to v2.6 and follow our formatting rules: https://github.com/PufferPanel/templates/blob/v2.6/README.md#formattng-rules
> 
> Please also confirm there is not a corrected template in the new branch.

I rebased the pr to v2.6 and updated the file to comply with the formatting rules and tested the template locally for good measure.
I also confirm that the template is not yet in the new branch.

I hope it can be merged soon.

**Comment by LordRalex** [comment]
Your PR is targeting the v2 branch, which needs to be the v2.6 branch.

**Comment by rickiewars** [comment]
How do I change this target branch? Or must I create a new pull-request for this?

**Comment by LordRalex** [comment]
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

**Comment by rickiewars** [comment]
Changed the base to v2.6
Please ignore my created pull request #196 (closed that one already)

File: minecraft-quilt/minecraft-quilt.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      "value": "1.19.3"
+    },
+    "ip": {
+      "type": "string",
+      "desc": "What IP to bind server to",
+      "display": "IP",
+      "required": true,
+      "value": "0.0.0.0"
+    },
+    "java": {
```
Comment (LordRalex) [inline minecraft-quilt/minecraft-quilt.json]: As of v2.6, we use the javadl option to handle the java requirements.
See https://github.com/PufferPanel/templates/blob/v2.6/minecraft-vanilla/minecraft-vanilla.json#L52-L55 and https://github.com/PufferPanel/templates/blob/v2.6/minecraft-vanilla/minecraft-vanilla.json#L73 for example usages.

File: minecraft-quilt/minecraft-quilt.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      "target": "mods",
+      "type": "mkdir"
+    },
+    {
+      "target": "installer",
+      "type": "mkdir"
+    },
+    {
+      "source": "quilt-installer.jar",
+      "target": "installer/quilt-installer.jar",
```
Comment (LordRalex) [inline minecraft-quilt/minecraft-quilt.json]: Any reason to keep the installer instead of deleting it?

**Review by LordRalex — CHANGES_REQUESTED** [review]

File: minecraft-quilt/minecraft-quilt.json (hunk 2 above)
Comment (rickiewars) [reply minecraft-quilt/minecraft-quilt.json]: No can be deleted indeed. This file was based on the Fabric config and it was done over there previously. That's why I kept it in

**Comment by rickiewars** [comment]
I'm not sure how terraria-tshock/terraria-tshock.json got edited, is that change correct?

**Comment by LordRalex** [comment]
That file is from v2, and not v2.6's file. Not sure if it's good though or not. If you can revert the changes, it'd be better.

**Comment by rickiewars** [comment]
I reverted that file and applied your feedback.
Also tested the template again on local server.

File: minecraft-quilt/minecraft-quilt.json [hunk 3]
```diff
@@ -0,0 +1,118 @@
+{
+  "name": "undefined",
```
Comment (LordRalex) [inline minecraft-quilt/minecraft-quilt.json]: Name is no longer defined correctly

File: minecraft-quilt/minecraft-quilt.json [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  ],
+  "run": {
+    "stop": "stop",
+    "command": "java17 -Xmx${memory}M -Dlog4j2.formatMsgNoLookups=true -jar quilt-server.jar nogui",
+    "workingDirectory": "",
+    "pre": [],
+    "post": [],
+    "environmentVars": {}
+  },
+  "data": {
```
Comment (LordRalex) [inline minecraft-quilt/minecraft-quilt.json]: Please put the data object after display, per https://github.com/PufferPanel/templates#formattng-rules

**Review by LordRalex — CHANGES_REQUESTED** [review]

**Comment by rickiewars** [comment]
Applied your feedback.

Conserning the formatting rules:

> I think everything was shuffled again whin I saved the config in Pufferfish for testing. Some lines needed changing so out of conveniance, I copied the working config directly out of PufferFish. I think there is an improvement here for Pufferfish to show the saved template to the user according to the formatting rules.

**Pull request merged**